test/test_gradient.jl

```diff
@@ -63,16 +67,37 @@ algs = [
     sierra_lite_diffusion,
 ]
 
-# Run tests using conversion to linear color space
-for alg in algs
-    println("")
-    print_braille(alg(img; to_linear=true); title="$(alg)")
+for alg in algs_deterministic
+    dither = alg(img)
+    @test_reference "references/grad_$(alg).txt" dither
+
+    dither = alg(img; to_linear=true)
+    @test_reference "references/grad_$(alg)_linear.txt" dither
+
+    # Visualize in terminal
+    print_braille(dither; title="$(alg)")
 end
 
-for level in 1:3
-    println("")
-    print_braille(
-        bayer_dithering(img; to_linear=true, level=level);
-        title="bayer_dithering, level $(level)",
-    )
+for level in 2:4
+    dither = bayer_dithering(img; level=level)
+    @test_reference "references/grad_bayer_dithering_l$(level).txt" dither
```

Comment:
> The reference test grad_bayer_dithering_l2.txt seems to fail because the returned BitMatrix consists of true/false rather than 1/0

ReferenceTest compares array data in string format :sad:... and `string(rand(Bool, 4, 4))` is different in Julia-1.0 and Julia-1.6...

For the correctness test, we could perhaps disable it in old Julia versions. Or, we could convert the data format into something that has consistent `string` result, e.g.,:

```suggestion
    @test_reference "references/grad_bayer_dithering_l$(level).txt" Int.(dither)
```